frontend/public/components/environment.jsx

```diff
@@ -21,16 +22,31 @@ const NameValueEditorComponent = (props) => <AsyncComponent loader={() => import
  * @private
  */
 const getPairsFromObject = (element) => {
```

Comment:
A description of the array it returns and what each value is would really help here.

Reply:
Just wondering if there's a reason not to use an object instead of an array here? Then you could name the properties instead of making them indexes in the array.

(Not necessarily suggesting you change that in this PR.)

Reply:
This is returning an object. I'll add more description in the comments.

Reply:
Right, but each property has a two-dimensional array. I was wondering if it should be an array of objects like

```
env: [{ name: envName, value: envValue, id: i}, ...]
```

Then you don't have to rely on knowing what is in each index of the array or mixing up the order for since they will be named properties.

(Not for this PR)